pkg/reconcile/example_test.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcile_test
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// This example implements a simple no-op reconcile function that prints the object to be Reconciled.
+func ExampleFunc() {
+	r := reconcile.Func(func(o reconcile.Request) (reconcile.Result, error) {
+		// Create your business logic to create, update, delete objects here.
+		fmt.Printf("Name: %s, Namespace: %s", o.Name, o.Namespace)
+		return reconcile.Result{}, nil
+	})
+
+	r.Reconcile(reconcile.Request{NamespacedName: types.NamespacedName{Namespace: "default", Name: "test"}})
+
+	// Output: Name: test, Namespace: default
+}
+
+// This example declares a simple type that implements reconcile.
+func ExampleReconcile() {
+	type MyReconcileImplementation struct {
```

Comment:
IMO, default example for demonstrating `Reconcile` should be struct based not adapter based because  99% use-cases involve a struct where an instance of `client` needs to be embedded etc. and speaking from my experience, I have found Adapter pattern a bit difficult to grok for newbies (experience from net/http)

Reply:
agreed